pkg/providers/imagefamily/image.go

```diff
@@ -77,39 +77,40 @@ func NewProvider(kubernetesInterface kubernetes.Interface, kubernetesVersionCach
 	}
 }
 
-// Get returns Distro and Image ID for the given instance type. Images may vary due to architecture, accelerator, etc
-func (p *Provider) Get(ctx context.Context, nodeClass *v1alpha2.AKSNodeClass, instanceType *cloudprovider.InstanceType, imageFamily ImageFamily) (string, string, error) {
-	defaultImages := imageFamily.DefaultImages()
-	for _, defaultImage := range defaultImages {
-		if err := instanceType.Requirements.Compatible(defaultImage.Requirements, v1alpha2.AllowUndefinedWellKnownAndRestrictedLabels); err == nil {
-			imageID, imageRetrievalErr := p.GetLatestImageID(ctx, defaultImage)
-			return defaultImage.Distro, imageID, imageRetrievalErr
+// TODO (charliedmcb): refactor this into resolver.go
+// resolveNodeImage returns Distro and Image ID for the given instance type. Images may vary due to architecture, accelerator, etc
+func (r *defaultResolver) resolveNodeImage(nodeImages []v1alpha2.NodeImage, instanceType *cloudprovider.InstanceType) (string, error) {
+	for _, availableImage := range nodeImages {
+		if err := instanceType.Requirements.Compatible(
+			scheduling.NewNodeSelectorRequirements(availableImage.Requirements...),
```

Comment:
Confirmed this translation back and forth of requirements should be fine. AWS does it as well.